pyproject.toml

```diff
@@ -35,6 +35,7 @@ grpcio = {version = "^1.48.2", optional = true}
 opentelemetry-api = { version = "^1.11.1", optional = true }
 opentelemetry-sdk = { version = "^1.11.1", optional = true }
 protobuf = ">=3.20"
+pydantic = { version = ">=1.10.0,<3.0.0", optional = true }
```

Comment:
Hrmm, so I suggested the `>=1.x` because I thought we were going to support v1 with our contrib converter. But now that it sounds like we're not, might as well do a `2.x` version constraint correct? Or does this version constraint somehow apply even when not using the extra?

Reply:
Ah I thought a `^2.x` bound would cause pip's dependency resolver to reject attempts to install `1.x` along with `temporalio`. Changed to
```toml
pydantic = { version = "^2.0.0", optional = true }
```

Reply:
>  I thought we were going to support v1 with our contrib converter. But now that it sounds like we're not

Yes, I backed out of that because (a) I think v2 is widely used already, (b) we want people to use v2 because v1 doesn't work with our sandbox, and (c) not a huge challenge but we'd have to write a backport of `to_jsonable_python` for v1, or similar shim.

Reply:
Works for me